app/src/main/java/me/phh/treble/app/Lid.kt

```diff
@@ -26,6 +26,35 @@ object Lid: EntryStartup {
         PowerManager::class.java.getMethod("goToSleep", Long::class.java).invoke(powerManager, SystemClock.uptimeMillis())
     }
 
+    fun huawei(ctxt: Context) {
+        val sensorManager = ctxt.getSystemService(SensorManager::class.java)
+        // Huawei Hall sensor for magnetic cover : ID = 65538
+        val sensors = listOf("HALL sensor")
```

Comment:
The name of the sensor isn't supposed to be stable
Could you rather use the type string rather than the name?

Reply:
Yes I also have the type (it is identical on all the kirin that I was able to test), I can change, I took the name because on the other brands, it was the name that was there. I will also add a return